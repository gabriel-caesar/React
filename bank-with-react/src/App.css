@import 'tailwindcss';

* {
  font-family: Tahoma;
}

p {
  font-family: Fira Code;
}