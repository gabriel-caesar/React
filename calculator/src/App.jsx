import './index.css';
import { useReducer } from 'react';
import DigitButton from './DigitButton';
import OperationButton from './OperationButton';

export const ACTIONS = {
  ADD_DIGIT: 'add-digit',
  CLEAR: 'clear',
  EVALUATE: 'evaluate',
  DELETE_DIGIT: 'delete-digit',
  CHOOSE_OPERATION: 'choose-operation',
};

function reducer(state, { type, payload }) {
  switch (type) {
    case ACTIONS.ADD_DIGIT:
      if (state.overwrite) {
        return {
          ...state,
          overwrite: false,
          currentOperand: payload.digit,
        };
      }
      if (payload.digit === '0' && state.currentOperand === '0') return state;
      if (payload.digit === '.' && state.currentOperand.includes('.'))
        return state;
      return {
        ...state, // state.currentOperand is there to maintain old values as new ones get output
        currentOperand: `${state.currentOperand || ''}${payload.digit}`,
      };
    case ACTIONS.DELETE_DIGIT:
      if (state.currentOperand !== 'undefined' || state.currentOperand !== '') {
        const reference = state.currentOperand.split('');
        reference.pop();
        return {
          ...state,
          currentOperand: reference.join(''),
        };
      }
      return state;
    case ACTIONS.CHOOSE_OPERATION:
      if (state.currentOperand == null && state.previousOperand == null)
        return state;
      if (state.previousOperand == null) {
        return {
          currentOperand: null,
          operator: payload.operation,
          previousOperand: state.currentOperand,
        };
      }
      if (state.currentOperand == null)
        return { ...state, operator: payload.operation };
      return {
        operator: payload.operation,
        currentOperand: null,
        previousOperand: evaluate(state),
      };
    case ACTIONS.CLEAR:
      return {};
    case ACTIONS.EVALUATE:
      if (state.currentOperand === undefined || state.currentOperand === null)
        return state;
      return {
        overwrite: true,
        operator: null,
        currentOperand: evaluate(state),
        previousOperand: null,
      };
  }
}

function evaluate(state) {
  const current = parseFloat(state.currentOperand);
  const previous = parseFloat(state.previousOperand);
  if (!isNaN(current) && !isNaN(previous)) {
    let calculation;
    switch (state.operator) {
      case '+':
        calculation = current + previous;
        break;
      case '-':
        calculation = current - previous;
        break;
      case '/':
        calculation = previous / current;
        break;
      case '*':
        calculation = current * previous;
        break;
    }

    return calculation
      .toFixed(2)
      .toString()
      .replace(/^[^0-9]/, '');
  }
}

export default function App() {
  const [{ currentOperand, previousOperand, operator }, dispatch] = useReducer(
    reducer,
    {}
  );

  return (
    <div className="calculator-grid">
      <div className="output">
        <div className="previous-operand">
          {previousOperand}
          {operator}
        </div>
        <div className="current-operand">{currentOperand}</div>
      </div>

      <div className="row">
        <button
          className="span-two"
          onClick={() => dispatch({ type: ACTIONS.CLEAR })}
        >
          AC
        </button>
        <button onClick={() => dispatch({ type: ACTIONS.DELETE_DIGIT })}>
          DEL
        </button>
        <OperationButton operation="/" dispatch={dispatch} />
      </div>
      <div className="row">
        <DigitButton digit="1" dispatch={dispatch} />
        <DigitButton digit="2" dispatch={dispatch} />
        <DigitButton digit="3" dispatch={dispatch} />
        <OperationButton operation="*" dispatch={dispatch} />
      </div>
      <div className="row">
        <DigitButton digit="4" dispatch={dispatch} />
        <DigitButton digit="5" dispatch={dispatch} />
        <DigitButton digit="6" dispatch={dispatch} />
        <OperationButton operation="+" dispatch={dispatch} />
      </div>
      <div className="row">
        <DigitButton digit="7" dispatch={dispatch} />
        <DigitButton digit="8" dispatch={dispatch} />
        <DigitButton digit="9" dispatch={dispatch} />
        <OperationButton operation="-" dispatch={dispatch} />
      </div>
      <div className="row">
        <DigitButton digit="." dispatch={dispatch} />
        <DigitButton digit="0" dispatch={dispatch} />
        <button
          className="span-two"
          onClick={() => dispatch({ type: ACTIONS.EVALUATE })}
        >
          =
        </button>
      </div>
    </div>
  );
}
