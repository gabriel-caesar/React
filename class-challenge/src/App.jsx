import FunctionalInput from './components/FunctionalInput.jsx';
import ClassInput from './components/ClassInput.jsx';
import './index.css';

export default function App() {
  return (
    <>
      <FunctionalInput name="Functional component!" />
      <div className="divider" />
      <ClassInput name="Class based component!" />
    </>
  );
}
