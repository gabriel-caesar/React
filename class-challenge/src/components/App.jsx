// App.jsx

const App = () => <h1 className='fancy' data-testid='this-header'>Mama</h1>;

export default App;
