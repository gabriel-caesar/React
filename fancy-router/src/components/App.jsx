import Buggs from './Buggs';
import { useState } from 'react';
import { useNavigate } from 'react-router-dom';

const App = () => {

  const navigate = useNavigate();

  return (
    <div className='bg-gray-900 h-screen flex flex-col justify-center items-center'>
      <h1 className='text-4xl text-white font-bold'>Fancy Router</h1>
      <p className='my-8 text-xl text-gray-400 underline'>Check these profiles out:</p>
      <div className='flex'>
        <button onClick={() => navigate('buggs/buggsDesc')} className='bg-gray-600 shadow rounded-md hover:cursor-pointer transition-colors hover:bg-blue-800 text-white p-2 font-bold text-xl'>Buggs Bunny</button>
      </div>
    </div>
  );
};

export default App;
