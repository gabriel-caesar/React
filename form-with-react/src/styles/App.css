* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
}

:root {
  --theme-color:#363636;
  --main-font-color: #dddddd;
  --dark-font-color: #c6c6c6;
  --highlight-color: #dab27a;
  --header-color: #6a764d;
}

h1 {
  font-size: 2rem;
  margin-bottom: 25px;
  text-align: center;
  color: var(--header-color);
}

input,
textarea {
  padding: 5px;
  font-size: 1.1rem;
  background-color: var(--dark-font-color);
  border: none;
  border-radius: 6px;
  color: #000;
}

label {
  display: flex;
  flex-direction: column;
  font-size: 1.2rem;
  color: var(--highlight-color)
}

label:not(:last-child) {
  margin-bottom: 15px;
}

section {
  margin-bottom: 30px;
  border-bottom: 1px solid var(--dark-font-color);
  padding-bottom: 30px;
  max-width: 443px;
}

button {
  background-color: var(--header-color);
  margin: auto;
  border: none;
  color: var(--main-font-color);
  padding: 10px 200px;
  font-size: 1.5rem;
  border-radius: 10px;
  margin-bottom: 55px;
  transition: opacity 0.15s;
}

button:hover {
  opacity: 0.7;
  cursor: pointer;
}

button:active {
  opacity: 0.3;
  transition: none;
}

.reviewApplication {
  font-size: 2rem;
}

p {
  font-size: 1.4rem;
  margin-bottom: 4px;
  width: auto;
  word-wrap: break-word;
}

span {
  font-weight: bold;
  color: var(--highlight-color);
  font-size: 1.5rem;
}