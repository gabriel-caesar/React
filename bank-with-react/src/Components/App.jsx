import { useState } from 'react';
import reactiveLogo from '../assets/reactive-no-bg.png';
import Button from './Buttons';
import CreateAccount from './CreateAccount';
import moneyDollar from '../assets/hand-with-money.svg';
import Login from './Login';
import '../App.css';

export default function App() {
  const [newUser, setNewUser] = useState({ username: '', password: '' });

  return (
    <div>
      <nav className="flex bg-green-900 justify-between items-center">
        <div className="flex relative w-100">
          <img
            src={reactiveLogo}
            alt="logo-image"
            className="w-28 box-border"
          />
          <p className="text-gray-200 absolute top-17 left-23 font-extrabold text-2xl">
            eactive Banking{' '}
          </p>
        </div>
        <CreateAccount state={newUser} setter={setNewUser} />
      </nav>

      <div className="flex justify-around items-center mt-16">
        <div className="flex items-center justify-center">
          <img
            src={moneyDollar}
            alt="hand full of dollar bills"
            className="w-80"
          />

          <h1 className="text-6xl w-100">
            New <span className="text-green-900 underline">Reactive</span>{' '}
            checking customers earn{' '}
            <span className="font-bold text-green-900">$300</span>
          </h1>
        </div>
        <Login user={newUser} />
      </div>
      <footer></footer>
    </div>
  );
}
