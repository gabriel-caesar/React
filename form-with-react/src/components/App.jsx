import { useState } from 'react'
import { Input, TextArea } from './Inputs'
import { Button } from './Button'
import { format } from 'date-fns'
import '../styles/App.css'
import '../styles/index.css'

function App() {
  const [isFilled, setIsFilled] = useState(false);
  const [applicant, setApplicant] = useState({
    fullName: '',
    email: '',
    phoneNumber: '',
    schoolName: '',
    studied: '',
    startedDateSchool: '',
    finishedDateSchool: '',
    companyName: '',
    positionTitle: '',
    responsabilities: '',
    startedDateJob: '',
    finishedDateJob: ''
  });

  function handleSubmit(e) {
    e.preventDefault();
    console.log('being clicked!', isFilled)
    return setIsFilled(true);
  };

  function handleDates(date) {
    if (date === '') {
      return date = '';
    } else {
      return format(new Date(date), 'MM/dd/yyyy');
    }
  };

  if (isFilled) {
    return (
      <>
        <p className='documentTitle'>CV Application</p>
        <h1 className='reviewApplication'>Review your Application</h1>

        <div className="reviewContainer">
          <section>
            <p><span>Full Name</span>: {applicant.fullName}</p>
            <p><span>Email</span>: {applicant.email}</p>
            <p><span>Phone Number</span>: {applicant.phoneNumber}</p>
          </section>

          <section>
            <p><span>School Name</span>: {applicant.schoolName}</p>
            <p><span>Title of Study</span>: {applicant.studied}</p>
            <p><span>Started Date</span>: {handleDates(applicant.startedDateSchool)}</p>
            <p><span>Finished Date</span>: {handleDates(applicant.finishedDateSchool)}</p>
          </section>

          <section>
            <p><span>Company Name</span>: {applicant.companyName}</p>
            <p><span>Position Title</span>: {applicant.positionTitle}</p>
            <p className='responsabilities'><span>Responsabilities</span>: {applicant.responsabilities}</p>
            <p><span>Started Date</span>: {handleDates(applicant.startedDateJob)}</p>
            <p><span>Finished Date</span>: {handleDates(applicant.finishedDateJob)}</p>
          </section>

        </div>
        <Button
          text='Edit'
          onClick={() => setIsFilled(false)}
        />
      </>
    )
  } else {
    return (
      <>
        <p className='documentTitle'>CV Application</p>
        <form
          onSubmit={handleSubmit}
        >
          <section>
            <h1>General Information</h1>
  
            <Input 
              label='Full Name'
              type='text'
              value={applicant.fullName}
              onChange={(e) => {
                const update = e.target.value;
                setApplicant({...applicant, fullName: update});
              }}
            />
  
            <Input 
              label='Email'
              type='email'
              value={applicant.email}
              onChange={(e) => {
                const update = e.target.value;
                setApplicant({...applicant, email: update});
              }}
            />
  
            <Input 
              label='Phone Number'
              type='number'
              value={applicant.phoneNumber}
              onChange={(e) => {
                const update = e.target.value;
                setApplicant({...applicant, phoneNumber: update});
              }}
            />
          </section>
  
          <section>
              <h1>Educational Experience</h1>
              
              <Input 
                label='School Name'
                type='text'
                value={applicant.schoolName}
                onChange={(e) => {
                  const update = e.target.value;
                  setApplicant({...applicant, schoolName: update});
                }}
              />
  
              <Input 
                label='Title of Study'
                type='text'
                value={applicant.studied}
                onChange={(e) => {
                  const update = e.target.value;
                  setApplicant({...applicant, studied: update});
                }}
              />
  
              <div className='datesWrapper'>
                <Input 
                  label='Started in'
                  type='date'
                  value={applicant.startedDateSchool}
                  onChange={(e) => {
                    const update = e.target.value;
                    setApplicant({...applicant, startedDateSchool: update});
                  }}
                />
  
                <Input 
                  label='Finished in'
                  type='date'
                  value={applicant.finishedDateSchool}
                  onChange={(e) => {
                    const update = e.target.value;
                    setApplicant({...applicant, finishedDateSchool: update});
                  }}
                />
              </div>
          </section>
  
          <section>
            <h1>Professional Experience</h1>
  
            <Input 
                label='Company Name'
                type='text'
                value={applicant.companyName}
                onChange={(e) => {
                  const update = e.target.value;
                  setApplicant({...applicant, companyName: update});
                }}
              />
  
              <Input 
                label='Position Title'
                type='text'
                value={applicant.positionTitle}
                onChange={(e) => {
                  const update = e.target.value;
                  setApplicant({...applicant, positionTitle: update});
                }}
              />
  
              <TextArea
                label='Professional responsabilities'
                value={applicant.responsabilities}
                onChange={(e) => {
                  const update = e.target.value;
                  setApplicant({...applicant, responsabilities: update});
                }}
                placeholder='At least 20 characters...'
              />
  
            <div className='datesWrapper'>
                <Input 
                  label='Started in'
                  type='date'
                  value={applicant.startedDateJob}
                  onChange={(e) => {
                    const update = e.target.value;
                    setApplicant({...applicant, startedDateJob: update});
                  }}
                />
  
                <Input 
                  label='Finished in'
                  type='date'
                  value={applicant.finishedDateJob}
                  onChange={(e) => {
                    const update = e.target.value;
                    setApplicant({...applicant, finishedDateJob: update});
                  }}
                />
              </div>
          </section>
          <Button
            text='Submit'
          />
        </form> 
      </>
    )
  }; // end of else
}

export default App
