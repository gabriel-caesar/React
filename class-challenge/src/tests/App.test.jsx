import { describe, it, expect } from "vitest";
import { render, screen, cleanup } from "@testing-library/react";
import App from "../components/App";

afterEach(cleanup);

describe("App component", () => {
  it("renders correct heading", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});

it('tests header', () => {
  const { getByTestId, getByText } = render(<App />)
  expect(getByTestId('this-header')).toHaveTextContent('Mama');
  expect(getByText('Mama')).toBeInTheDocument();
  expect(getByText('Mama')).toHaveClass('fancy');
})