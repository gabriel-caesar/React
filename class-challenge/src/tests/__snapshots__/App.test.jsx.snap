// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders correct heading 1`] = `
<DocumentFragment>
  <h1
    class="fancy"
    data-testid="this-header"
  >
    Mama
  </h1>
</DocumentFragment>
`;
